src/wpantund/NCPInstanceBase.cpp

```diff
@@ -843,6 +854,7 @@ NCPInstanceBase::handle_ncp_state_change(NCPState new_ncp_state, NCPState old_nc
 	// (Special case of InterfaceUp -> InterfaceDown)
 	} else if (ncp_state_is_commissioned(old_ncp_state)
 		&& !ncp_state_is_commissioned(new_ncp_state)
+		&& !ncp_state_is_joining(new_ncp_state)
```

Comment:
The condition here checks if old state was commissioned and new state is not, then we go ahead and issue `reset_interface()` which then erases all addresses/prefixes/etc. The main use-case is when for example, we "leave", to clear all the table and reset everything.

The new extra `ncp_state_is_joining()` is to avoid the following case:
- A already commissioned NCP is reset
- We initialize it and we end up in "COMMISIONED" state
- Then we resume, this causes a transition to "ASSOCIATING" state (for which `ncp_state_is_commissioned()` correctly returns `false`)
- This causes the addresses/routes/etc to be wiped.